src/fmelt.c

```diff
@@ -265,7 +268,7 @@ SEXP checkVars(SEXP DT, SEXP id, SEXP measure, Rboolean verbose) {
       tmp = PROTECT(unlist_(tmp2)); protecti++;
     }
     for (i=0; i<length(tmp); i++) {
-      if (INTEGER(tmp)[i] <= 0 || INTEGER(tmp)[i] > ncol)
+      if (invalid_measure(INTEGER(tmp)[i], ncol)) 
```

Comment:
here is a fix for the case of id.vars NOT null, which just repeats the same logic as used for the case for id.vars IS null.